### ISSUE TITLE
Cannot pair with Apple TV 4 on Windows: "Keep-alive not supported"

### ISSUE THREAD

**Issue by briankendall** [postlund/pyatv#1881]
### Describe the bug

I'm trying to use `atvremote` to pair with my Apple TV 4 running tvOS 12.4 using the MRP protocol.

`atvremote` can successfully scan the network and find my Apple TV with `atvremote scan`. However the pair command fails. It displays a warning "Keep-alive not supported" and throws exceptions. I am also unable to pair using the API, and get the same warning and exceptions if I try.

### Error log

After executing the pair command, the following gets printed:
```
2022-11-21 23:14:16 WARNING [pyatv.protocols.mrp.connection]: Keep-alive not supported: Option TCP_KEEPIDLE is not supported
2022-11-21 23:14:21 ERROR [pyatv.scripts.atvremote]: Pairing failed
Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\pyatv\protocols\mrp\protocol.py", line 274, in _receive
    await semaphore.acquire()
  File "C:\Python310\lib\asyncio\locks.py", line 390, in acquire
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\pyatv\support\__init__.py", line 33, in error_handler
    return await func(*args, **kwargs)
  File "C:\Python310\lib\site-packages\pyatv\protocols\mrp\auth.py", line 46, in start_pairing
    resp = await self.protocol.send_and_receive(msg, generate_identifier=False)
  File "C:\Python310\lib\site-packages\pyatv\protocols\mrp\protocol.py", line 261, in send_and_receive
    return await self._receive(identifier, timeout)
  File "C:\Python310\lib\site-packages\pyatv\protocols\mrp\protocol.py", line 273, in _receive
    async with async_timeout.timeout(timeout):
  File "C:\Python310\lib\site-packages\async_timeout\__init__.py", line 129, in __aexit__
    self._do_exit(exc_type)
  File "C:\Python310\lib\site-packages\async_timeout\__init__.py", line 212, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\pyatv\scripts\atvremote.py", line 179, in pair
    await self._perform_pairing(pairing)
  File "C:\Python310\lib\site-packages\pyatv\scripts\atvremote.py", line 189, in _perform_pairing
    await pairing.begin()
  File "C:\Python310\lib\site-packages\pyatv\protocols\mrp\pairing.py", line 51, in begin
    return await error_handler(
  File "C:\Python310\lib\site-packages\pyatv\support\__init__.py", line 35, in error_handler
    raise exceptions.ConnectionFailedError(str(ex)) from ex
pyatv.exceptions.ConnectionFailedError
```

### How to reproduce the bug?

It fails when trying to pair with my Apple TV like so:

`atvremote --id [apple TV ID here] --protocol mrp pair`

### What is expected behavior?

It pairs with the Apple TV.

### Operating System

Windows 8.1

### Python

3.10

### pyatv

0.10.3

### Device

Apple TV 4th gen, tvOS 12.4

### Additional context

I've tried using `atvremote` to pair with the same Apple TV in python 3.9 on macOS and did not have this issue.

**Comment by briankendall** [comment]
Looks like `TCP_KEEPIDLE` is only supported in Windows 10 1709 and later. That would explain why this doesn't work.

**Comment by briankendall** [comment]
I'm not sure what happened, but suddenly the latest version of pyatv is working correctly on my system.

`TCP_KEEPIDLE` warning is apparently irrelevant and does not prevent pyatv or `atvremote` from pairing. I'm not sure why it wasn't working before, but between when I made my original bug report and now, I uninstalled and reinstalled several different versions of pyatv to experiment and see which one works the best. Now that I've reinstalled the latest version, things are working. I can only presume that something about doing that kicked my system out of a non-working state and into a working state. What more I cannot say.
